### PR TITLE
Spanish NIE (Foreigner Resident Identity) recognizer

### PR DESCRIPTION
**Pull request by areyesfalcon**
## Change Description

Added a new Recognizer for Spanish Foreigners Resident Identity (NIE)

## Issue reference

This PR doesn't fix any issue

## Checklist

- [X] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [X] I have signed the CLA (if required)
- [X] My code includes unit tests
- [X] All unit tests and lint checks pass locally
- [X] My PR contains documentation updates / additions if required

### CODE REVIEW

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/es_nie_recognizer.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    """
+    Recognize NIE number using regex and checksum.
+
+    :param patterns: List of patterns to be used by this recognizer
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    :param replacement_pairs: List of tuples with potential replacement values
+    for different strings to be used during pattern matching.
+    This can allow a greater variety in input, for example by removing dashes or spaces.
```
Comment (omri374) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/es_nie_recognizer.py:16]: Can you please add a link to the source of the regex and checksum logic as a comment?

Comment (areyesfalcon) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/es_nie_recognizer.py:16]: Of course:

- https://es.wikipedia.org/wiki/N%C3%BAmero_de_identidad_de_extranjero
- https://www.interior.gob.es/opencms/ca/servicios-al-ciudadano/tramites-y-gestiones/dni/calculo-del-digito-de-control-del-nif-nie/
- https://www.juntadeandalucia.es/servicios/madeja/contenido/libro-pautas/196
- https://www.juntadeandalucia.es/servicios/madeja/contenido/recurso/677
- https://www.policia.es/_es/extranjeria_extranjeros.php
- https://www.boe.es/buscar/act.php?id=BOE-A-1997-3364
- https://www.boe.es/buscar/doc.php?id=BOE-A-2008-12050
-

Comment (omri374) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/es_nie_recognizer.py:16]: Thanks! Please add one of those (say the second link) in a code comment

**Comment by areyesfalcon** [comment]
> @areyesfalcon please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by areyesfalcon** [comment]
@microsoft-github-policy-service agree

**Comment by areyesfalcon** [comment]
I've added references of the ES_NIE algorithm

**Review by omri374 — APPROVED** [review]
Thanks! great addition

**Comment by omri374** [comment]
@areyesfalcon there are a few linting issues. Could you please fix those by running flake8? Please write here if you need any support. Thanks!

**Comment by omri374** [comment]
Hi @areyesfalcon, would you like us to help with resolving the linting issues?

**Pull request closed without merging**